src/main/java/seedu/address/logic/commands/DeleteContactedInfoCommand.java

```diff
@@ -0,0 +1,161 @@
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.reminder.ReminderList;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Deletes a specified contacted info from an existing person in the address book, if list is not empty.
+ */
+public class DeleteContactedInfoCommand extends Command {
+    public static final String COMMAND_WORD = "delContacted";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes a contacted information from an existing contact, "
+            + "as specified by the index number used in the displayed person list. The tag will be deleted only if "
+            + "the contacted information list is not empty\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_DELETE_CONTACTED_INFO + "INDEX\n"
+            + "Example: " + COMMAND_WORD + " 2 "
+            + PREFIX_DELETE_CONTACTED_INFO + "1";
+
+    public static final String MESSAGE_DELETE_CONTACTED_INFO_SUCCESS = "Deleted contacted information: %1$s";
+    public static final String MESSAGE_MISSING_INFO = "Index specified out of range.";
+    public static final String MESSAGE_EMPTY_LIST = "Cannot delete, list is empty!";
+
+    private final Index index;
+    private final int indexToDel;
+
+    private ContactedInfo contactedInfo;
+    /**
+     * Constructs an {@code DeleteContactedInfoCommand} with the given {@code Index} and {@code ContactedInfo}.
+     *
+     * @param index of the person in the filtered person list to add the {@code Tag}.
+     * @param indexToDel  to be deleted from the {@code Person} specified by {@code index}.
+     */
+    public DeleteContactedInfoCommand(Index index, int indexToDel) {
+        requireNonNull(index);
+        requireNonNull(indexToDel);
+
+        this.index = index;
+        this.indexToDel = indexToDel - 1;
+    }
+
+    /**
+     * Deletes a tag from an existing person in the address book, if the tag exists.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return the command result after the command execution.
+     * @throws CommandException if the {@code Tag} do not exist or the {@code Index} is invalid.
+     */
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        if (personToEdit.getContactedInfoList().isEmpty()) {
+            throw new CommandException(MESSAGE_EMPTY_LIST);
+        }
+
+        if (!(personToEdit.getContactedInfoList().size() > indexToDel)) {
```

Comment:
This seems too complicated, you can consider doing:
`if (indexToDel >= personToEdit.getContactedInfoList().size())`

Reply:
But on top of that, `personToEdit.getContactInfoList().size()` is not slap, so you should make a method in `Person`. See `getFilteredListSize()`